### PR TITLE
Update field access completion sorting order to prioritize record/object fields

### LINKED ISSUE #20313: Completion suggestions for records first displays the functions
Refer to the following:
Peek 2019-12-10 10-52
For records, I feel that it'd be more useful to display the fields of the records first because that's what the users will be working with most of the time.

### PR DESCRIPTION
**Pull request by IMS94**
## Purpose
$subject

Fixes #20313

## Approach
If a completion item is a `record field` or an `object field` while the node at cursor is a `field access`, we now show the record/object fields at the top of the list. Overridden the `sort()` method in `FieldAccessContext` to achieve this.

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [x] Added necessary tests
  - [x] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [x] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/FieldAccessContext.java [hunk 1]
```diff
@@ -52,4 +53,16 @@ public FieldAccessContext(Class<T> attachmentPoint) {
 
         return this.getCompletionItemList(symbolList, ctx);
     }
+
+    @Override
+    public void sort(BallerinaCompletionContext context, T node, List<LSCompletionItem> completionItems) {
+        // First we do the default sorting
+        super.sort(context, node, completionItems);
```
Comment (nadeeshaan) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/providers/context/FieldAccessContext.java:60]: Here we traverse the completion item list twice. In this particular context we can traverse the list once and assign the priorities. or, in a more configurable way, we can add a sort API to the sorting utils to accept a priority offset and assign the priority a completion item by offsetting the default priority.

**Review by nadeeshaan — APPROVED** [review]

**Review by nadeeshaan — CHANGES_REQUESTED** [review]
Here we traverse the completion item list twice. In this particular context we can traverse the list once and assign the priorities. or, in a more configurable way, we can add a sort API to the sorting utils to accept a priority offset and assign the priority a completion item by offsetting the default priority.

**Comment by IMS94** [comment]
@nadeeshaan Updated the PR as we discussed in a meeting to use a utility method to get the rank of a completion item with an optional offset.

**Review by nadeeshaan — APPROVED** [review]
LGTM

**Pull request merged**